src/lib/BUILD.gn

```diff
@@ -13,9 +13,10 @@
 # limitations under the License.
 
 import("//build_overrides/chip.gni")
+import("${chip_root}/src/platform/device.gni")
 import("lib.gni")
 
-static_library("lib") {
+source_set("base") {
```

Comment:
name "base" fills like a filler name (same like support/core/helper/util/common ... unfortunately we have a lot of those). How about "without-logging"

then target is `src/lib:without-logging` andlib adds logging to without-logging.